packages/purpose-process/src/services/purposeService.ts

```diff
@@ -334,3 +393,107 @@ const replacePurposeVersion = (
     updatedAt: newVersion.updatedAt,
   };
 };
+
+const getInvolvedTenantByEServiceMode = async (
+  eservice: EService,
+  consumerId: TenantId,
+  readModelService: ReadModelService
+): Promise<Tenant> => {
+  if (eservice.mode === eserviceMode.deliver) {
+    return retrieveTenant(consumerId, readModelService);
+  } else {
+    return retrieveTenant(eservice.producerId, readModelService);
+  }
+};
+
+const updatePurposeInternal = async (
```

Comment:
Minor: may I suggest to rename this? `performUpdatePurpose`, `executeUpdatePurpose`...